compiler/codegen/src/compile.rs

```diff
@@ -1275,6 +1275,17 @@ impl Compiler {
 
         let prev_class_name = std::mem::replace(&mut self.class_name, Some(name.to_owned()));
 
+        // Check if the class is declared global
+        let symbol_table = self.symbol_table_stack.last().unwrap();
+        let symbol = symbol_table.lookup(name.as_ref()).expect(
+            "The symbol must be present in the symbol table, even when it is undefined in python.",
+        );
+        let mut prefix_path = vec![];
+        if symbol.scope == SymbolScope::GlobalExplicit {
+            while let Some(value) = self.qualified_path.pop() {
+                prefix_path.insert(0, value);
+            }
+        }
```

Comment:
Because `prefix_path` is only used for global, I'd like to add it to variable names.
I also suggest changing the while loop to extend and drain.
```suggestion
        let mut global_path_prefix = Vec::new();
        if symbol.scope == SymbolScope::GlobalExplicit {
            global_path_prefix.extend(self.qualified_path.drain(..));
        }
```

Reply:
extend & drain is new to me! Thanks

Reply:
I found that clippy suggests using `append`

https://rust-lang.github.io/rust-clippy/master/index.html#extend_with_drain

Reply:
that looks better